### PR TITLE
[UPDATED] - MR:completed user registartion,login using jwt token , al…

### PR DESCRIPTION
**Pull request by Jinal-81**
…so completed the user listing.

### CODE REVIEW

File: user/__pycache__/urls.cpython-38.pyc
Comment (abhishek145607) [inline user/__pycache__/urls.cpython-38.pyc:1]: Remove all .pyc file from commit

File: user/api/serializers.py [hunk 1]
```diff
@@ -1,14 +1,76 @@
...
+
+class RegisterSerializer(serializers.ModelSerializer):
+    email = serializers.EmailField(
+        required=True,
+        validators=[UniqueValidator(queryset=User.objects.all())], max_length=30
+    )
+
+    password = serializers.CharField(write_only=True, required=True, validators=[validate_password])
+    password2 = serializers.CharField(write_only=True, required=True)
+    first_name = serializers.CharField(
```
Comment (abhishek145607) [inline user/api/serializers.py:27]: Not necessary first name and last name characters.

File: user/api/constants.py
Comment (abhishek145607) [inline user/api/constants.py:1]: Constants.py defined in utils file and messages format should be:

```
class AuthConstant:

// constant messges
```

**Review by abhishek145607 — CHANGES_REQUESTED** [review]

File: spotify/settings/base.py [hunk 2]
```diff
@@ -253,8 +253,7 @@
 
 REST_FRAMEWORK = {
     "DEFAULT_AUTHENTICATION_CLASSES": (
-        "rest_framework.authentication.SessionAuthentication",
-        "rest_framework.authentication.TokenAuthentication",
```
Comment (yatishTrootech) [inline spotify/settings/base.py:257]: take default permission class

File: user/api/constants.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+PASSWORD_NOT_MATCH_ERROR_MSG = "Password fields didn't match."
```
Comment (yatishTrootech) [inline user/api/constants.py]: PASSWORD_DOES_NOT_MATCH

File: user/api/constants.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+FIRST_NAME_CHAR_ONLY_ERROR_MSG = 'First name should only contain characters.'
```
Comment (yatishTrootech) [inline user/api/constants.py]: don't write MSG

File: user/api/serializers.py [hunk 5]
```diff
@@ -1,14 +1,76 @@
...
 
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ["username", "email", "name", "url"]
+        fields = ('username', 'email', 'first_name', 'last_name')
 
+
+class RegisterSerializer(serializers.ModelSerializer):
```
Comment (yatishTrootech) [inline user/api/serializers.py]: inherit UserSerializer

File: user/api/serializers.py [hunk 6]
```diff
@@ -1,14 +1,76 @@
...
+    class Meta:
+        model = User
+        fields = ('username', 'password', 'password2', 'email', 'first_name', 'last_name')
         extra_kwargs = {
-            "url": {"view_name": "user_api:user-detail", "lookup_field": "username"}
+            'first_name': {'required': True},
+            'last_name': {'required': True}
         }
+
+    def validate(self, attrs):
```
Comment (yatishTrootech) [inline user/api/serializers.py]: validate_password

File: user/api/serializers.py [hunk 7]
```diff
@@ -1,14 +1,76 @@
...
+        return attrs
+
+    def create(self, validated_data):
+        user = User.objects.create(
+            username=validated_data['username'],
+            email=validated_data['email'],
+            first_name=validated_data['first_name'],
+            last_name=validated_data['last_name']
+        )
+
```
Comment (yatishTrootech) [inline user/api/serializers.py:58]: User.objects.create(**validated_data)

Comment (yatishTrootech) [reply user/api/serializers.py:58]: or super.craete()

File: user/api/views.py [hunk 8]
```diff
@@ -1,24 +1,27 @@
...
-from .serializers import UserSerializer
+from .serializers import RegisterSerializer, MyTokenObtainPairSerializer, UserSerializer
 
 User = get_user_model()
 
 
-class UserViewSet(RetrieveModelMixin, ListModelMixin, UpdateModelMixin, GenericViewSet):
-    serializer_class = UserSerializer
+class RegisterView(generics.CreateAPIView):
     queryset = User.objects.all()
```
Comment (yatishTrootech) [inline user/api/views.py]: User.objects.none()

File: user/api/views.py [hunk 9]
```diff
@@ -1,24 +1,27 @@
...
-    def me(self, request):
-        serializer = UserSerializer(request.user, context={"request": request})
-        return Response(status=status.HTTP_200_OK, data=serializer.data)
+class MyObtainTokenPairView(TokenObtainPairView):
+    permission_classes = (AllowAny,)
+    serializer_class = MyTokenObtainPairSerializer
+
+
+class UserListView(ListAPIView):
+    queryset = User.objects.all()
```
Comment (yatishTrootech) [inline user/api/views.py]: active()

File: user/api/views.py [hunk 10]
```diff
@@ -1,24 +1,27 @@
...
+    permission_classes = (IsAuthenticated,)
```
Comment (yatishTrootech) [inline user/api/views.py]: remove permission_classes

File: user/templates/signup.html [hunk 11]
```diff
@@ -40,42 +40,42 @@ <h2 class="fw-bold mb-5">Sign up now</h2>
...
+                    <input type="email" id="form3Example4" class="form-control" />
+                    <label class="form-label" for="form3Example2">Email</label>
+                  </div>
+                </div>
               </div>
 
               <!-- Password input -->
               <div class="form-outline mb-4">
-                <input type="password" id="form3Example4" class="form-control" />
+                <input type="password" id="form3Example5" class="form-control" />
```
Comment (yatishTrootech) [inline user/templates/signup.html]: proper id name form3Example5

**Review by yatishTrootech — CHANGES_REQUESTED** [review]

File: user/api/serializers.py (hunk 1 above)
Comment (Jinal-81) [reply user/api/serializers.py:27]: yes, for the practice purpose added here

File: user/api/constants.py [hunk 12]
```diff
@@ -1,4 +1,9 @@
...
-FIRST_NAME_CHAR_ONLY_ERROR_MSG = 'First name should only contain characters.'
-LAST_NAME_CHAR_ONLY_ERROR_MSG = 'Last name should only contain characters.'
-REGEX_FOR_CHAR_ONLY = r'^[a-zA-Z]+$'
\ No newline at end of file
+
+class AuthConstantsMessages:
+    """
+    class that contains constant for User Authentication messages
+    """
+    PASSWORD_DOES_NOT_MATCH_ERROR_MESSAGE = "Password fields didn't match."
```
Comment (abhishek145607) [inline user/api/constants.py:6]: Remove ERROR_MESSAGE

**Review by abhishek145607 — APPROVED** [review]

**Pull request merged**